### ISSUE TITLE
`axis`/`dims` default value in `concatenate`

### ISSUE THREAD

**Issue by lostanlen** [kymatio/kymatio#903]
While rebasing PR #871 i noticed that our `concatenate` operator has strange default values for `dims`:

Here's `backend/numpy_backend.py`:
```python
    @classmethod
    def concatenate(cls, arrays, dim=1):
        return cls._np.stack(arrays, axis=dim)
```

Here's `backend/tensorflow_backend.py`:
```python
    @staticmethod
    def concatenate(arrays, dim=1):
        return tf.stack(arrays, axis=dim)
```

Here's `backend/torch_backend.py`:
``` python
   @staticmethod
    def concatenate(arrays, dim=2):
        return torch.stack(arrays, dim=dim)
```

I'm a bit confused by this `dim=2` default value. And not too happy that the kwargs is called `dim` instead of `axis`, which would be more compliant with the Python Array API. 

Our decision here will affect the semantics of #902 

@eickenberg  thoughts?
